### PR TITLE
Fix tests in CI

### PR DESCRIPTION
**Pull request by MatthewPortman**
### Description

This pull request is to address several failures in CI. Tickets have been created for tests that are newly skipped. Other tests have been re-arranged to isolate failures.

I have also removed a duplicate label test in `jdaviz/configs/imviz/tests/test_parser.py` since the app behavior has changed when loading identical data (see changes in `jdaviz/configs/imviz/tests/test_footprints.py` as well).

Jessie's [GWCS PR](https://github.com/spacetelescope/jdaviz/pull/4032) fixes the `TypeError: The value must be a valid Python or Numpy numeric type.` issue. Once that's merged and this is rebased, expect all tests to pass (barring any remote tomfoolery).

### Change log entry

- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] If new remote data is added that uses MAST, is the URI added to the `cache-download.yml` workflow?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone. Bugfix milestone also needs an accompanying backport label.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: jdaviz/configs/imviz/tests/test_parser.py [hunk 1]
```diff
@@ -351,16 +351,6 @@ def test_parse_jwst_nircam_level2(self, imviz_helper):
...
-            # Test duplicate label functionality
-            imviz_helper.app.data_collection.clear()
-            imviz_helper.load_data(pf, ext='SCI', data_label='TEST', show_in_viewer=False)
-            data = imviz_helper.app.data_collection[0]
-            assert data.label.endswith('[SCI,1]')
-
-            imviz_helper.load_data(pf, ext='SCI', data_label='TEST', show_in_viewer=False)
-            data = imviz_helper.app.data_collection[1]
-            assert data.label.endswith('[SCI,1] (1)')
-
```
Comment (kecnry) [inline jdaviz/configs/imviz/tests/test_parser.py:363]: why is this being removed?

File: jdaviz/configs/specviz2d/plugins/spectral_extraction/tests/test_spectral_extraction.py [hunk 2]
```diff
@@ -511,19 +511,22 @@ def test_labels_and_spectral_extraction_flux_difference(self, method, helper, re
...
         # Another 2D Spectral Extraction <=> 2D Spectrum [spectral axis]
-        # Another 2D Spectral Extraction <=> 2D Spectrum
-        assert len(dc.external_links) == 9
-        for link in dc.external_links:
-            # Check that linking is correct by confirming that both
-            # are in `expected_labels`
-            assert (link.data1.label in expected_labels) and (link.data2.label in expected_labels)
-            assert isinstance(link, LinkSameWithUnits)
+        # Another 2D Spectral Extraction <=> 2D Spectrum)
+        # TODO: Investigate number of links in py313 dev tests
```
Comment (kecnry) [inline jdaviz/configs/specviz2d/plugins/spectral_extraction/tests/test_spectral_extraction.py:528]: is there a ticket for this or intended to do as part of this effort yet?

File: jdaviz/core/loaders/tests/test_loaders.py [hunk 3]
```diff
@@ -218,7 +222,7 @@ def test_resolver_url(deconfigged_helper):
 
     # may change with future importers
     assert len(loader.format.choices) == 3
-    assert loader.format.selected == 'Image'  # default may change with future importers
+    assert loader.format.selected == '2D Spectrum'
```
Comment (kecnry) [inline jdaviz/core/loaders/tests/test_loaders.py]: I'm not sure that this is 100% predictable right now, so may warrant a follow-up ticket still.  Let's also keep the comment though so we know that a change to the default isn't necessarily a break, but just a behavior change

File: jdaviz/core/loaders/tests/test_loaders.py [hunk 4]
```diff
@@ -231,7 +235,7 @@ def test_resolver_url(deconfigged_helper):
     assert loader.importer.data_label == 'exnkul627fcuhy5akf2gswytud5tazmw_index-0'  # noqa
 
     loader.target = 'Any'
-    assert len(loader.format.choices) == 4
+    assert len(loader.format.choices) == 3
```
Comment (kecnry) [inline jdaviz/core/loaders/tests/test_loaders.py:238]: what format was removed here?

**Review by kecnry — COMMENTED** [review]
just a few questions - overall the changes seem reasonable to me as long as we prioritize the follow-up tickets to unskip the checks you skip here

File: jdaviz/core/loaders/tests/test_loaders.py (hunk 4 above)
Comment (MatthewPortman) [reply jdaviz/core/loaders/tests/test_loaders.py:238]: I'm pretty sure it's a result of strengthening the restrictions on 2d spectra from [this PR](https://github.com/spacetelescope/jdaviz/pull/3996), so we lost Image as an option which also explains the line(s) above this one.

File: jdaviz/configs/specviz2d/plugins/spectral_extraction/tests/test_spectral_extraction.py (hunk 2 above)
Comment (MatthewPortman) [reply jdaviz/configs/specviz2d/plugins/spectral_extraction/tests/test_spectral_extraction.py:528]: Yes I made a ticket and [Gilbert's PR](https://github.com/spacetelescope/jdaviz/pull/3983) touches it as well.

File: jdaviz/configs/imviz/tests/test_parser.py (hunk 1 above)
Comment (MatthewPortman) [reply jdaviz/configs/imviz/tests/test_parser.py:363]: I'm not exactly sure when this happened but we/something changed when we try to load duplicate data. It appears to overwrite the original data so we don't get two entries in the data collection (it fails on `data = imviz_helper.app.data_collection[1]`). And if I try to load in the same data with a different label, the assert statement fails. So, as long as the overwriting is intentional, I figured that this part of the test isn't serving the purpose it was intended anymore. Also I'm pretty sure we test a duplicate label somewhere else that does work...

I ran into the same overwrite issue with `jdaviz/configs/imviz/tests/test_footprints.py`, hence my changes there. Here we don't use both after the fact, it's just this one check.

Comment (MatthewPortman) [reply jdaviz/configs/imviz/tests/test_parser.py:363]: I just looked into it, this might be an Imviz bug not a feature. We have several tests for duplicate labels (`jdaviz/tests/test_app.py`) but they all seem to use specviz and those are passing.

That being said, if it is indeed a bug, I say we add the data label test in `test_app.py` and remove it from here to be consistent with those tests and keep things modular.

File: jdaviz/core/loaders/tests/test_loaders.py (hunk 3 above)
Comment (MatthewPortman) [reply jdaviz/core/loaders/tests/test_loaders.py]: I've restored the comment, it'll be pushed with the next batch of commits.

File: jdaviz/core/loaders/tests/test_loaders.py [hunk 5]
```diff
@@ -122,6 +122,10 @@ def test_markers_specviz2d_unit_conversion(specviz2d_helper, spectrum2d):
 
 @pytest.mark.remote_data
 @pytest.mark.filterwarnings(r"ignore::astropy.wcs.wcs.FITSFixedWarning")
+@pytest.mark.xfail(reason='spectral_axis unit failure is due to a temporary fix'
```
Comment (cshanahan1) [inline jdaviz/core/loaders/tests/test_loaders.py:125]: is there a ticket for this with instructions to un xfail this test?

File: jdaviz/configs/imviz/tests/test_footprints.py [hunk 6]
```diff
@@ -22,8 +22,8 @@ def test_user_api(imviz_helper, image_2d_wcs, tmp_path):
     arr = np.ones((10, 10))
     ndd = NDData(arr, wcs=image_2d_wcs)
     # load the image twice to test linking
-    imviz_helper.load_data(ndd)
-    imviz_helper.load_data(ndd)
+    imviz_helper.load_data(ndd, data_label='data1')
```
Comment (cshanahan1) [inline jdaviz/configs/imviz/tests/test_footprints.py:25]: why do these need specific data labels set? does this test not run with the automatically generated labels

File: jdaviz/core/loaders/tests/test_loaders.py (hunk 5 above)
Comment (MatthewPortman) [reply jdaviz/core/loaders/tests/test_loaders.py:125]: Yes, I referenced it in the upstream glue ticket so whenever that goes in, we can come back and remove this.

File: jdaviz/configs/imviz/tests/test_footprints.py (hunk 6 above)
Comment (MatthewPortman) [reply jdaviz/configs/imviz/tests/test_footprints.py:25]: I think it's an Imviz bug I erroneously thought was a feature (ha!), see one of my comments above -- https://github.com/spacetelescope/jdaviz/pull/4036#discussion_r2834544052

Having the duplicate label isn't necessary for this test though so even when we fix the bug, I don't think this needs to be reverted.

File: jdaviz/configs/imviz/tests/test_parser.py (hunk 1 above)
Comment (MatthewPortman) [reply jdaviz/configs/imviz/tests/test_parser.py:363]: OK I looked into it some more... 
* `test_duplicate_data_labels` as it is, which uses specviz with `load_data`, does not fail. 
* When I change to using imviz, it fails. 
* When I try to set the same test up with deconfigged but with `load` for imviz, specviz, and specviz2d, they all fail. 
* When trying the same test using only the UI, we get duplicate labels as expected.

**Review by cshanahan1 — APPROVED** [review]

File: jdaviz/configs/imviz/tests/test_parser.py (hunk 1 above)
Comment (MatthewPortman) [reply jdaviz/configs/imviz/tests/test_parser.py:363]: For now, I've updated a test in `test_app` to test duplicate labels in multiple configs and marked with `xfail`.

**Review by kecnry — APPROVED** [review]
